tests/test_scalars.py

```diff
@@ -0,0 +1,713 @@
+    check(assert_type(dt.datetime(2012, 1, 1) - td, dt.datetime), dt.datetime)
+    check(assert_type(dt.date(2012, 1, 1) - td, dt.date), dt.date)
+    check(assert_type(np.datetime64(1, "ns") - td, pd.Timestamp), pd.Timestamp)
+    check(assert_type(dt.timedelta(days=1) - td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(np.timedelta64(1, "D") - td, pd.Timedelta), pd.Timedelta)
+    check(
+        assert_type(
+            pd.period_range("2012-01-01", periods=3, freq="D") - td, pd.PeriodIndex
+        ),
+        pd.PeriodIndex,
+    )
+    check(
+        assert_type(pd.date_range("2012-01-01", periods=3) - td, pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(np.array([1, 2, 3], dtype="timedelta64[D]") - td, np.ndarray),
+        np.ndarray,
+    )
+    check(
+        assert_type(np.array([1, 2, 3], dtype="datetime64[ns]") - td, np.ndarray),
+        np.ndarray,
+    )
+
+    check(assert_type(td * 3, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td * 3.5, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td * np.array([1, 2, 3]), np.ndarray), np.ndarray)
+    check(assert_type(td * np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
+    check(assert_type(td * pd.Series([1, 2, 3]), pd.Series), pd.Series)
+    check(assert_type(td * pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
+    check(assert_type(td * pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(
+        assert_type(td * pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
+        pd.TimedeltaIndex,
+    )
+
+    check(assert_type(td // td, int), int)
+    check(assert_type(td // 3, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td // 3.5, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td // np.array([1, 2, 3]), np.ndarray), np.ndarray)
+    check(assert_type(td // np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
+    check(assert_type(td // pd.Series([1, 2, 3]), pd.Series), pd.Series)
+    check(assert_type(td // pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
+    check(assert_type(td // pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(
+        assert_type(td // pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
+        pd.TimedeltaIndex,
+    )
+
+    check(assert_type(td / td, float), float)
+    check(assert_type(td / 3, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td / 3.5, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td / np.array([1, 2, 3]), np.ndarray), np.ndarray)
+    check(assert_type(td / np.array([1.2, 2.2, 3.4]), np.ndarray), np.ndarray)
+    check(assert_type(td / pd.Series([1, 2, 3]), pd.Series), pd.Series)
+    check(assert_type(td / pd.Series([1.2, 2.2, 3.4]), pd.Series), pd.Series)
+    check(assert_type(td / pd.Index([1, 2, 3]), pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(
+        assert_type(td / pd.Index([1.2, 2.2, 3.4]), pd.TimedeltaIndex),
+        pd.TimedeltaIndex,
+    )
```

Comment:
for both division operators, check them as `something / td`, etc

Reply:
The reverse divisions are missing because they error.